domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/fareattributes/FareAttribute.java

```diff
@@ -0,0 +1,158 @@
+    @NotNull
+    public String getCurrencyType() {
+        return currencyType;
+    }
+
+    @NotNull
+    public PaymentMethod getPaymentMethod() {
+        return paymentMethod;
+    }
+
+    @NotNull
+    public Transfers getTransfers() {
+        return transfers;
+    }
+
+    @Nullable
+    public String getAgencyId() {
+        return agencyId;
+    }
+
+    @Nullable
+    public Integer getTransferDuration() {
+        return transferDuration;
+    }
+
+    public FareAttribute(@NotNull final String fareId,
+                         final float price,
+                         @NotNull final String currencyType,
+                         @NotNull final PaymentMethod paymentMethod,
+                         @NotNull final Transfers transfers,
+                         @Nullable final String agencyId,
+                         @Nullable final Integer transferDuration) {
+        this.fareId = fareId;
+        this.price = price;
+        this.currencyType = currencyType;
+        this.paymentMethod = paymentMethod;
+        this.transfers = transfers;
+        this.agencyId = agencyId;
+        this.transferDuration = transferDuration;
+    }
+
+    public static class FareAttributeBuilder {
+        private String fareId;
+        private Float price;
+        private String currencyType;
+        private PaymentMethod paymentMethod;
+        private Transfers transfers;
+        private String agencyId;
+        private Integer transferDuration;
+
+        public FareAttributeBuilder fareId(String fareId) {
+            this.fareId = fareId;
+            return this;
+        }
+
+        public FareAttributeBuilder price(Float price) {
+            this.price = price;
+            return this;
+        }
+
+        public FareAttributeBuilder currencyType(String currencyType) {
+            this.currencyType = currencyType;
+            return this;
+        }
+
+        public FareAttributeBuilder paymentMethod(Integer paymentMethod) {
+            this.paymentMethod = PaymentMethod.fromInt(paymentMethod);
+            return this;
+        }
+
+        public FareAttributeBuilder transfers(Integer transfers) {
+            this.transfers = Transfers.fromInt(transfers);
+            return this;
+        }
+
+        public FareAttributeBuilder agencyId(@Nullable String agencyId) {
+            this.agencyId = agencyId;
+            return this;
+        }
+
+        public FareAttributeBuilder transferDuration(int transferDuration) {
+            this.transferDuration = transferDuration;
+            return this;
+        }
+
+        public FareAttribute build() {
+            if (fareId == null) {
+                throw new IllegalArgumentException("field `fare_id` in file `fare_attributes.txt` can not be null");
+            }
+            if (price == null) {
+                throw new IllegalArgumentException("field `price` in file `fare_attributes.txt` can not be null");
+            }
+            if (currencyType == null) {
```

Comment:
We should at some point be validating `currencyType` as a valid ISO 4217 alphabetical currency code.

One way to do this is try call `Currency.getInstance(currencyType)`, and if an `IllegalArgumentException` is thrown then it's not valid.

See https://docs.oracle.com/javase/8/docs/api/java/util/Currency.html#getInstance-java.lang.String-

Reply:
Yes. I suggest to have a dedicated `ValidateCurrencyCode` use case that will be run after we are done filling the GtfsDataRepository

Reply:
That's one possible approach. I see this as a data type validation, though. Aren't we attempting to validate all data types prior to filling the repo with the record?

Reply:
Could that be done in class GtfsTypeValidator, at the same place where other gtfs types are checked?

Reply:
Good point. In my head data validation concerned what can be found in the schema (ie: range for numeric types) and this is more advanced, but I was mistaken as that is not even true for timezone validity which is a more advanced case not treated through a use case. So yes you are rigtht @lionel-nj, the correct place for it is in GtfsTypeValidator.

Reply:
Comment: closing this conversation when closing https://github.com/MobilityData/gtfs-validator/issues/123

Reply:
A proposition for currency code validation have been implemented here: https://github.com/MobilityData/gtfs-validator/pull/127